src/Dialect/Krnl/KrnlHelper.hpp

```diff
@@ -154,12 +154,19 @@ class BuildKrnlLoop {
 public:
   // Create kernel loop builder for a loop nest of depth loopNum.
   BuildKrnlLoop(ConversionPatternRewriter &rewriter, Location loc, int loopNum);
+  BuildKrnlLoop(const BuildKrnlLoop &) = delete;
+  BuildKrnlLoop(BuildKrnlLoop &&) = delete;
+  BuildKrnlLoop &operator=(const BuildKrnlLoop &) = delete;
+  BuildKrnlLoop &operator=(BuildKrnlLoop &&) = delete;
 
   // Create kernel loop builder for a loop nest of depth equal to the
   // dimensionality of the operand. An operand of MemRef type is requied.
   BuildKrnlLoop(
       ConversionPatternRewriter &rewriter, Location loc, Value memRefOperand);
-  ~BuildKrnlLoop();
+  virtual ~BuildKrnlLoop() {
```

Comment:
Confused by the virtual, esp since there appears to be no subclasses.

Reply:
I have been trained to shiver at the sight of a `virtual`... esp for cases where execution time is paramount. It is less important here, I would go with the final then.